src/samples/plugins/ModularSessionKeyPlugin.sol

```diff
@@ -0,0 +1,338 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.19;
+
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {UserOperation} from "@eth-infinitism/account-abstraction/interfaces/UserOperation.sol";
+import {UpgradeableModularAccount} from "../../account/UpgradeableModularAccount.sol";
+import {
+    ManifestFunction,
+    ManifestAssociatedFunctionType,
+    ManifestAssociatedFunction,
+    PluginManifest,
+    PluginMetadata,
+    SelectorPermission
+} from "../../interfaces/IPlugin.sol";
+import {BasePlugin} from "../../plugins/BasePlugin.sol";
+import {ISessionKeyPlugin} from "./interfaces/ISessionKeyPlugin.sol";
+import {ISingleOwnerPlugin} from "../../plugins/owner/ISingleOwnerPlugin.sol";
+import {SingleOwnerPlugin} from "../../plugins/owner/SingleOwnerPlugin.sol";
+import {PluginStorageLib, StoragePointer} from "../../libraries/PluginStorageLib.sol";
+
+/// @title Modular Session Key Plugin
+/// @author Decipher ERC-6900 Team
+/// @notice This plugin allows some designated EOA or smart contract to temporarily
+/// own a modular account.
+/// This modular session key plugin acts as a 'parent plugin' for all specific session
+/// keys. Using dependency, this plugin can be thought as a parent contract that stores
+/// session key duration information, and validation functions for session keys. All
+/// logics for session keys will be implemented in child plugins.
+/// It allows for session key owners to access MSCA both through user operation and
+/// runtime, with its own validation functions.
+/// Also, it has a dependency on SingleOwnerPlugin, to make sure that only the owner of
+/// the MSCA can add or remove session keys.
```

Comment:
Could you add a note here and in the other plugin that these plugins are for demonstrating ERC-6900 capabilities only, and should not serve as examples of production-ready session key plugins?

Reply:
Added it!